gamify-specs/src/main/resources/api-spec.yaml

```diff
@@ -59,6 +59,66 @@ paths:
         '201':
           description: created
 
+  /categories:
+    post:
+      description: post a new category
+      operationId: addCategory
+      requestBody:
+        content:
+          'application/json':
+            schema:
+              $ref: '#/components/schemas/Category'
+      responses:
+        '201':
+          description: created
+    get:
+      description: read a category
+      operationId: getCategory
+      parameters:
+        - name: idCategory
+          in: query
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: success
+          content:
+            'application/json':
+              schema:
+                $ref: '#/components/schemas/Category'
+    put:
+      description: update a category
+      operationId: updateCategory
+      parameters:
+        - name: idCategory
+          in: query
+          required: true
+          schema:
+            type: string
+      requestBody:
+        content:
+          'application/json':
+            schema:
+              $ref: '#/components/schemas/Category'
+      responses:
+        '201':
+          description: created
+    delete:
+      description: delete a category
+      operationId: deleteCategory
+      parameters:
+        - name: idCategory
+          in: query
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: success
```

Comment:
Do we want to have this kind of architecture or the `/categories/id` type ? I'm in favour of the second version, but we should discuss it with everybody